### PR TITLE
FIX: ensure that events are integers

### PR DESCRIPTION
**Pull request by kingjr**
See https://github.com/mne-tools/mne-python/issues/2406#issuecomment-131577612

### CODE REVIEW

**Comment by kingjr** [comment]
@Eric89GXL @agramfort ready for review. Not sure whether the test shouldn't be in _BaseEpochs directly?

**Comment by Eric89GXL** [comment]
Yeah the fix should be in `_BaseEpochs` since this could be an issue for any kind of Epochs object created

**Comment by wmvanvliet** [comment]
could you also create a unit test for this?

File: mne/epochs.py [hunk 1]
```diff
@@ -178,6 +178,11 @@ def __init__(self, info, data, events, event_id, tmin, tmax,
         del event_id
 
         if events is not None:  # RtEpochs can have events=None
+
+            if not np.array_equal(events, np.array(events, dtype=int)):
+                raise ValueError('events values be integers.')
```
Comment (wmvanvliet) [inline mne/epochs.py]: events needs to be an array of type int

File: mne/epochs.py [hunk 2]
```diff
@@ -178,6 +178,11 @@ def __init__(self, info, data, events, event_id, tmin, tmax,
         del event_id
 
         if events is not None:  # RtEpochs can have events=None
+
+            if not np.array_equal(events, np.array(events, dtype=int)):
```
Comment (agramfort) [inline mne/epochs.py]: it's not the right way to do it. Use

```
if events.dtype.kind != 'i':
```

File: mne/epochs.py [hunk 3]
```diff
@@ -178,6 +178,11 @@ def __init__(self, info, data, events, event_id, tmin, tmax,
...
+                raise ValueError('events values be integers.')
+            events = np.array(events, dtype=int)
```
Comment (agramfort) [inline mne/epochs.py]: it's will be not needed. And it's btw not the standard way to change the dtype of an array. 

events = events.astype(int)

is the standard way.

**Comment by agramfort** [comment]
besides LGTM.

File: mne/epochs.py (hunk 2 above)
Comment (kingjr) [inline mne/epochs.py]: but is the users passes a float array that only contains integers, we don't want to throw an error do we?

Comment (agramfort) [inline mne/epochs.py]: for me yes we do.

Comment (wmvanvliet) [inline mne/epochs.py]: +1

**Comment by kingjr** [comment]
comments addressed, thanks!

**Comment by kingjr** [comment]
@agramfort @wmvanvliet being strict on events `dtype` (i.e. rejecting float array even if their values are only composed of integers) generates bugs everywhere in mne, because many scripts generate events like `np.zeros(n_trials, 3)` without specifying `dtype`.  I could fix it everywhere, but I think this is a good sign we shouldn't if we don't want to create a backward incompatibility.

**Comment by Eric89GXL** [comment]
Regardless of whether we change this check (I'm fine either way), we should
fix all those instances you mention, those are bugs

**Comment by agramfort** [comment]
let's try to fix indeed those bugs. We'll see how much we're likely to
break some code. Can you take care of it?

**Comment by kingjr** [comment]
Mmmh, no.

**Comment by Eric89GXL** [comment]
I can do it a couple hours from now, then

**Comment by kingjr** [comment]
thanks Eric

**Comment by kingjr** [comment]
If we go for the strict version, I think we'll need to add a deprecation warning, as ~~my~~ users' scripts will start crashing where they used not to.

**Comment by Eric89GXL** [comment]
closing for #2415

**Pull request closed without merging**